From 2f314cf5d94d39abe97b80733f8e630b37d70fa2 Mon Sep 17 00:00:00 2001
From: Radek Doulik <rodo@novell.com>
Date: Fri, 05 Jun 2009 08:51:35 +0000
Subject: fix impress crash related to fit to frame text feature

* patches/dev300/apply:
* patches/dev300/svx-fix-fit-to-frame-crash.diff:
---
diff --git a/patches/dev300/apply b/patches/dev300/apply
index 48c495d..baa6a0c 100644
--- a/patches/dev300/apply
+++ b/patches/dev300/apply
@@ -3042,10 +3042,10 @@ ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
 # Fix for crash on Windows by Rail Aliev. Root cause unknown.
 configmgr-validate-removedtree.diff, n#505704
 
+svx-fix-fit-to-frame-crash.diff, n#508621, rodo
+
 [ UnusedMethodsRemoval ]
 # Patches in this section remove unused methods identified by callcatcher.
 
 # NOTE! This section should always be at the bottom so that no other patches would
 # depend on it.
-
-
diff --git a/patches/dev300/svx-fix-fit-to-frame-crash.diff b/patches/dev300/svx-fix-fit-to-frame-crash.diff
new file mode 100644
index 0000000..9302a1c
--- a/dev/null
+++ b/patches/dev300/svx-fix-fit-to-frame-crash.diff
@@ -0,0 +1,12 @@
+diff -rup svx/source/editeng-orig/impedit3.cxx svx/source/editeng/impedit3.cxx
+--- svx/source/editeng-orig/impedit3.cxx	2009-06-05 10:19:00.000000000 +0200
++++ svx/source/editeng/impedit3.cxx	2009-06-05 10:45:28.000000000 +0200
+@@ -940,7 +940,7 @@ sal_Bool ImpEditEngine::CreateLines( USH
+ //						nCurPos -= rLRItem.GetTxtLeft();	// Tabs relativ zu LI
+ 						// Skalierung rausrechnen
+ 						if ( aStatus.DoStretch() && ( nStretchX != 100 ) )
+-							nCurPos = nCurPos*100/nStretchX;
++						    nCurPos = nCurPos*100/std::max(static_cast<sal_Int32>(nStretchX), static_cast<sal_Int32>(1));
+ 
+                         short nAllSpaceBeforeText = static_cast< short >(rLRItem.GetTxtLeft()/* + rLRItem.GetTxtLeft()*/ + nSpaceBeforeAndMinLabelWidth);
+                         aCurrentTab.aTabStop = pNode->GetContentAttribs().FindTabStop( nCurPos - nAllSpaceBeforeText /*rLRItem.GetTxtLeft()*/, aEditDoc.GetDefTab() );
--
cgit v0.8.2
